Update databricks-configs.md to include MV/ST

benc-db opened the pull request:
## What are you changing in this pull request and why?
<!---
Describe your changes and why you're making them. If linked to an open
issue or a pull request on dbt Core, then link to them here! 

To learn more about the writing conventions used in the dbt Labs docs, see the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md).
-->
Editing the Databricks configuration reference:

* Expanded configuration table to indicate that we now support all of the main table config options in Python as well as SQL, with the exception of Liquid Clustering, which is not yet part of PySpark.

* Added section describing support for materialized_view and streaming_table.

## Checklist
- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

benc-db committed:
Update databricks-configs.md to include MV/ST

* Expanded configuration table to indicate that we now support all of the main table config options in Python as well as SQL, with the exception of Liquid Clustering, which is not yet part of PySpark.

* Added section describing support for materialized_view and streaming_table.

benc-db commented:
Have a few more edits to make; reading through the style guide now.

benc-db committed:
Small changes to conform better to the style guide.

benc-db committed:
Using more active voice

benc-db committed:
Be clearer about a link destination

dataders commented on website/docs/reference/resource-configs/databricks-configs.md:
```diff
@@ -281,3 +281,45 @@ snapshots:
+
+## Advanced Materializations
+
+Starting with version 1.6.0, the dbt-databricks adapter supports [materialized views](https://docs.databricks.com/en/sql/user/materialized-views.html) and [streaming tables](https://docs.databricks.com/en/sql/load-data-streaming-table.html), as alternatives to incremental tables that are powered by (Delta Live Tables)[https://docs.databricks.com/en/delta-live-tables/index.html].
```
swap the `[` and `(`

```suggestion
Starting with version 1.6.0, the dbt-databricks adapter supports [materialized views](https://docs.databricks.com/en/sql/user/materialized-views.html) and [streaming tables](https://docs.databricks.com/en/sql/load-data-streaming-table.html), as alternatives to incremental tables that are powered by [Delta Live Tables](https://docs.databricks.com/en/delta-live-tables/index.html).
```

dataders commented on website/docs/reference/resource-configs/databricks-configs.md:
```diff
@@ -281,3 +281,45 @@ snapshots:
+## Advanced Materializations
+
+Starting with version 1.6.0, the dbt-databricks adapter supports [materialized views](https://docs.databricks.com/en/sql/user/materialized-views.html) and [streaming tables](https://docs.databricks.com/en/sql/load-data-streaming-table.html), as alternatives to incremental tables that are powered by (Delta Live Tables)[https://docs.databricks.com/en/delta-live-tables/index.html].
+Read their entries in the [Delta Live Tables documentation site](https://docs.databricks.com/en/delta-live-tables/index.html#what-are-delta-live-tables-datasets) to understand their use cases.
```
```suggestion
See [What are Delta Live Tables?](https://docs.databricks.com/en/delta-live-tables/index.html#what-are-delta-live-tables-datasets) for more information and use cases.
```

dataders commented on website/docs/reference/resource-configs/databricks-configs.md:
```diff
@@ -281,3 +281,45 @@ snapshots:
 </File>
+
+
+## Advanced Materializations
```
I say perhaps just call it what it is for the time being

```suggestion
## Materialized views and streaming stables
```

dataders commented on website/docs/reference/resource-configs/databricks-configs.md:
```diff
@@ -281,3 +281,45 @@ snapshots:
+
+When dbt detects a pre-existing relation of one of these types, it issues a `REFRESH` [command](https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-refresh-full.html).
+
+### Limitations
+
+As mentioned above, support for these materializations in the Databricks adapter is still limited.
+At this time the following configuration options are not available:
+
+* Specifying a refresh schedule for these materializations
+* Specifying `on_configuration_change` settings.
+  
+These configuration settings will be supported in the near future.
```
what about the scenario where i change the SQL statement in the model file and want to drop the old version and make a new one? will this happen? this isn't supported (yet) in our adapters, but we will support it soon

benc-db replied:
I suspect this will not work, similar to the call out on the Redshift documentation, but will test on Monday and update when I know.

dataders reviewed (changes requested):
thanks for the contribution @benc-db, much appreciated. I especially like your addition of a "model support" column to the configuration option table.

i left some small comments. the only other ask I have is to wrap the new MV/ST section in a `VersionBlock` so users of earlier versions of dbt Core aren't lead astray. Trivial example below

```md
<VersionBlock firstVersion="1.6">
chunk of markdown
</VersionBlock>
```

benc-db committed:
Apply suggestions from code review

Co-authored-by: Anders <anders.swanson@dbtlabs.com>

dataders committed:
Merge branch 'current' into patch-1

dataders committed:
Merge branch 'current' into patch-1

dataders committed:
Merge branch 'current' into patch-1

benc-db committed:
Updating MV/ST section

Updating to indicate this is a 1.6 feature with limitations that we plan to address in the 1.7 timeframe.

dataders commented:
> @dataders any suggestion on how to incorporate similar information into the model config table? Prior to 1.6.0, most of the configs were not supported for Python.

yo! great point. I've also wanted to have this ability to have rows show up according to the version (perhaps a secret markdown column that gets rendered out?). But the best we can do is to maintain two tables. Here's an example that I did recently

https://github.com/dbt-labs/docs.getdbt.com/blob/3dd58697521cda71eed7c329286742ae04888e46/website/docs/docs/build/incremental-models.md?plain=1#L254-L282

dataders committed:
Merge branch 'current' into patch-1

benc-db committed:
Update databricks-configs.md to indicate that 1.6 brought expanded python config support

benc-db committed:
Update databricks-configs.md to add documentation for replace_where

benc-db commented:
@dataders anything else you need from me?

dataders committed:
Merge branch 'current' into patch-1

dataders reviewed (approved):
@benc-db this is a huge level up for our docs. because there's so much here and it's been open for two weeks now, I'm going to ship it :shipit:. perhaps next time a PR like this could be 3 distinct ones so that they can get merged faster

benc-db commented:
> @benc-db this is a huge level up for our docs. because there's so much here and it's been open for two weeks now, I'm going to ship it :shipit:. perhaps next time a PR like this could be 3 distinct ones so that they can get merged faster

Acknowledged.  I probably should not have added the replace_where, but I couldn't believe I had found an incremental strategy that we shipped in 1.4 that still had no doc.  Will try to make future updates more focused.
